src/audio/component.c

```diff
@@ -70,7 +70,7 @@ struct comp_dev *comp_new(struct sof_ipc_comp *comp)
 		return NULL;
 	}
 
-	trace_event(TRACE_CLASS_COMP, "comp new %s type %d id %d.%d",
+	trace_event(TRACE_CLASS_COMP, "comp new %d type %d id %d.%d",
```

Comment:
My question is, we're just showing the component type as an integer? I'm not sure if "uid" can be correctly printed as an integer. We need to look into this further.

Reply:
https://github.com/thesofproject/sof/blob/af38b20bc84f79c81fc37066cdd255b81cb08b11/src/include/sof/audio/component.h#L393

Reply:
It's an address so this is not going to work correctly; it will differ from component to component. It may be fine for now but know that the "uid" number is not stable and you have to keep yourself in the loop when UUID-based logging becomes fully merged.

Reply:
Not gonna block on this though, just something to be aware of.

Reply:
OK, if this an address should this be a pointer?

Reply:
We can begin as a pointer, however UIDs are actually special and might warrant its own specifier in the future.

Reply:
@finikorg @paulstelian97 sof-logger converts special 0x1f.... 32bit uid value to string representation (full uuid value formatted plus name of the identified object) when %s type is specified in the format string. This change brakes the functionality. If you do not like the idea of implicit type conversion applied to arguments to match the % type (because other loggers do not support it I guess), please propose another light way for the current logging stack.

Reply:
OK, I forgot about log encoding. If this breaks sof-logger I will close the issue.